portfolio/src/main/webapp/script.js

```diff
@@ -1,28 +1,133 @@
+function getRandomPresident() {
+  document.getElementById('game_button').style.display = 'none';
+  let presidents = ['George Washington', 'John Adams', 'Thomas Jefferson', 
+  'James Madison', 'James Monroe', 'John Quincy Adams', 'Andrew Jackson', 
+  'Martin Van Buren', 'William Henry Harrison', 'John Tyler', 'James K. Polk',
+  'Zachary Taylor', 'Millard Fillmore', 'Franklin Pierce', 'James Buchanan', 
+  'Abraham Lincoln', 'Andrew Johnson', 'Ulysses S. Grant', 'Rutherford B. Hayes', 
+  'James A. Garfield', 'Chester A. Arthur', 'Grover Cleveland', 'Benjamin Harrison', 
+  'Grover Cleveland', 'William McKinley', 'Theodore Roosevelt', 'William Howard Taft',
+  'Woodrow Wilson', 'Warren G. Harding', 'Calvin Coolidge', 'Herbert Hoover', 
+  'Franklin D. Roosevelt', 'Harry S. Truman', 'Dwight D. Eisenhower', 'John F. Kennedy',
+  'Lyndon B. Johnson', 'Richard Nixon', 'Gerald Ford', 'Jimmy Carter', 'Ronald Reagan', 
+  'George H. W. Bush', 'Bill Clinton', 'George W. Bush', 'Barack Obama', 'Donald Trump'];
+  const president_answer_pos = Math.floor(Math.random() * presidents.length);
+  const president_answer = presidents[president_answer_pos];
+  localStorage.setItem('president_answer', president_answer);
+  presidents.splice(president_answer_pos, 1); 
+  let choices = makeChoices(presidents, president_answer, president_answer_pos);
+  updateQuestion(president_answer_pos);
+  updateOptions(choices);
 }
+/*picks a unique set of 4 presidents (including the proposed correct one) and returns them as an options array*/
+function makeChoices(presidents, president_answer, president_answer_pos){
+  const choice_pos = Math.floor(Math.random() * 4);
+  let choices = [];
+  let count = 0;
+  while(choices.length < 4){
+    if(count === choice_pos){
+      choices.push(president_answer);
+    }else{
+      const president_pos = Math.floor(Math.random() * presidents.length);
+      const president = presidents[president_pos];
+      presidents.splice(president_pos, 1); 
+      choices.push(president);
+    }
+    count++;
+  }
+  return choices;
+}
+/*takes president's position in array to implement the ordinal-based "who's the __th" president question format*/
+function updateQuestion(president_answer_pos){
+  const name_num = president_answer_pos + 1;
+  let name = name_num + '';
+  if(name_num % 10 === 1 && name_num !== 11){
+    name += 'st';
+  }
+  else if(name_num % 10 === 2 && name_num !== 12){
+    name += 'nd';
+  }
+  else if(name_num % 10 === 3 && name_num !== 13){
+    name += 'rd';
+  }
+  else{
+    name += 'th';
+  }
+  let question = document.getElementById('question');
+  question.style.display = 'block';
+  question.innerText = 'Who was the ' + name + ' president of the United States?';
+}
+/*Changes the button option texts and values so that the next question can be displayed*/
+function updateOptions(choices){
+  let one = document.getElementById('one');
```

Comment:
you could make this a loop over a {'one', 'two', 'three', 'four'} array since you're using the same logic for each one